### DISCUSSION TITLE
Multi-targeting breaks when PackageId is specified in a dependency

### DISCUSSION THREAD

**Discussion by Mrcubix** [dotnet/runtime#108343] [Q&A]
I have found out today that i cannot make use of multi-targeting if any of the underlying dependencies have a `PackageId` Property Declared (Along with a version).

For instance, let's say i have 2 projects, which only differ by the target framework & package version :
```csproj
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net5</TargetFramework> <!-- the other version uses net6.0 -->
    <LangVersion>preview</LangVersion>
    <Nullable>enable</Nullable>
  </PropertyGroup>

  <PropertyGroup>
    <PackageId>Some.Dependency</PackageId> <!-- Commenting fix the issue -->
    <Version>0.5.3.3</Version> <!-- the other version uses 0.6.4.0 -->
    <Authors>Gess1t</Authors>
    <Description>Some Dependency</Description>
    <PackageOutputPath>../nupkg</PackageOutputPath>
    <PackageLicenseExpression>LGPL-3.0-or-later</PackageLicenseExpression>
  </PropertyGroup>

</Project>
```

Now let's say i have a library that support multi-targeting like such :
```csproj
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFrameworks>net5.0;net6.0</TargetFrameworks>
    <LangVersion>preview</LangVersion>
    <Nullable>enable</Nullable>
  </PropertyGroup>

  <ItemGroup Condition=" '$(TargetFramework)' == 'net5.0' ">
    <ProjectReference Include="../Some.Dependency/Some.Dependency.csproj" Version="0.5.3.3" />
  </ItemGroup>

  <ItemGroup Condition=" '$(TargetFramework)' == 'net6.0' ">
    <ProjectReference Include="../Some.Dependency-0.6.x/Some.Dependency-0.6.x.csproj" Version="0.6.4.0" />
  </ItemGroup>

</Project>
```

If i now try to build an app using this Library in an app targeting `net5.0` then i get this error on compilation :
```log
error NU1201: Project Some.Dependency is not compatible with net5.0 (.NETCoreApp,Version=v5.0). Project 
Some.Dependency supports: net6.0 (.NETCoreApp,Version=v6.0)
```

if i look inside inside project.assets.json in the obj folder, then i can see this :
```json
"targets": {
    "net5.0": {
      "Some.Dependency/0.6.4": {
        "type": "project",
        "framework": "Unsupported,Version=v0.0",
        "compile": {
          "bin/placeholder/Some.Dependency.dll": {}
        },
        "runtime": {
          "bin/placeholder/Some.Dependency.dll": {}
        }
      }
    },
    "net6.0": {
      "Some.Dependency/0.6.4": {
        "type": "project",
        "framework": ".NETCoreApp,Version=v6.0",
        "compile": {
          "bin/placeholder/Some.Dependency.dll": {}
        },
        "runtime": {
          "bin/placeholder/Some.Dependency.dll": {}
        }
      }
    }
  },
  ```
  
For some reasons, Something in the restore process cause something to set the dependency in both target to the latest version of Some.Dependency, instead of the one specified in the condition.

Here is the repo used : https://github.com/Mrcubix/Multi-Targeting-Issue

Any ideas on how i could fix such an issue?

The current workaround i have right now is to create a separate folder for the .NET 6 target and include the source from the other Lib.
Also this obviously seems to be a bug, wondering if i should address that in the nuget repo, the sdk or here.

**Comment by huoyaoyuan** [comment]
Is it related to https://github.com/NuGet/Home/issues/10368 ?

**Reply by Mrcubix** [reply]
I do see error messages that are simillar, but most of the cases are between a package reference & a project reference, while here, both are a project reference.

However, it does seem like there are situations where they could be restored as packages instead of projects? Which would seem very odd to me.

Another edit : on a second note, this is about 2 packages with a different target framework but the same version, slightly different from my case.
